### PR TITLE
test: e2e improvements and new fmt/vet checks in Makefile

### PR DESCRIPTION
**Pull request by jpower432**
# Description

This PR moves all e2e test script and configurations to a central directory and breaks up `lib.sh` logically into multiple scripts. It also adds `go vet` and `go fmt` to the sanity check in the Makefile.

Blocked by #389 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [X] E2E tests updated

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: test/e2e/e2e-simple.sh
Comment (jharmison-redhat) [inline test/e2e/e2e-simple.sh:8]: I know that when run from the Makefile target these will source fine, but from any other directory they won't. Any chance you could do something like:

```shell
SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
source "$SCRIPT_DIR/lib/check.sh"
source "$SCRIPT_DIR/lib/workflow.sh"
source "$SCRIPT_DIR/lib/util.sh"
source "$SCRIPT_DIR/testcases.sh"
```

**Review by jharmison-redhat — COMMENTED** [review]
Looks fine to me. Some things shellcheck doesn't like but not things that will break things very badly.

**Comment by jharmison-redhat** [comment]
Holding approval until other merge and rebase, as discussed offline.

File: test/e2e/e2e-simple.sh
Comment (jpower432) [reply test/e2e/e2e-simple.sh:8]: Will do. Thanks!

**Review by jharmison-redhat — COMMENTED** [review]
Shell looks good, all of `shellcheck`'s remaining complaints are non-issues. Waiting on a Golanger to approve the `go vet`-induced changes.

`make test-e2e` execution itself is unaffected, and the organizational changes make sense.

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Review by dinhxuanvu — COMMENTED** [review]
/lgtm

**Pull request merged**
